tests/baselines/reference/templateLiteralTypesPatterns.types

```diff
@@ -646,7 +646,7 @@ conversionTest("testDowncast");
 >"testDowncast" : "testDowncast"
 
 function conversionTest2(groupName: | "downcast" | "dataDowncast" | "editingDowncast" | {} & `${string}Downcast`) {}
->conversionTest2 : (groupName: "downcast" | "dataDowncast" | "editingDowncast" | {} & `${string}Downcast`) => void
+>conversionTest2 : (groupName: "downcast" | "dataDowncast" | "editingDowncast" | ({} & `${string}Downcast`)) => void

```

Comment:
Just commenting on the ones that aren't whitespace only for visiblity

Reply:
I think this is because our parenthesizer when we clone the node is stricter than the syntax allows. I'll have to check.

Reply:
Yeah, now that we make a new node to apply `SingleLine` to the `{}`, the node goes through the parenthesizer, and the relevant rule is
```ts
    // UnionType[Extends] :
    //     `|`? IntersectionType[?Extends]
    //     UnionType[?Extends] `|` IntersectionType[?Extends]
    //
    // - A union type constituent has the same precedence as the check type of a conditional type
    function parenthesizeConstituentTypeOfUnionType(type: TypeNode) {
        switch (type.kind) {
            case SyntaxKind.UnionType: // Not strictly necessary, but a union containing a union should have been flattened
            case SyntaxKind.IntersectionType: // Not strictly necessary, but makes generated output more readable and avoids breaks in DT tests
                return factory.createParenthesizedType(type);
        }
        return parenthesizeCheckTypeOfConditionalType(type);
    }
```
which acknowledges it's extraneous.

Reply:
I do wonder if there's a still a reason to do it; the union case seems like overkill, and I recently changed ExpectType to normalize unions and could probably just make it always strip or add the parens.

Reply:
I sent https://github.com/microsoft/TypeScript/pull/57900 which shows the affect of this; I do sorta think we should stop doing this as I don't think that the extra parens are that much of an improvement?

But, on its own I think this particular change is alright.

Reply:
Personally I found it a bit easier to read with the parens, so consider that a soft vote for keeping the parens.

> I do wonder if there's a still a reason to do it; the union case seems like overkill

Are there cases of non-normalization to consider?